src/LibraryManager.Vsix/Commands/RestoreOnBuildCommand.cs

```diff
@@ -1,37 +1,38 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Design;
+using System.Linq;
 using EnvDTE;
-using Microsoft.Web.LibraryManager.Contracts;
 using Microsoft.VisualStudio.ComponentModelHost;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.Web.LibraryManager.Contracts;
+using Microsoft.Web.LibraryManager.Vsix.Contracts;
 using NuGet.VisualStudio;
-using System;
-using System.ComponentModel.Design;
-using System.Linq;
-using System.Collections.Generic;
 using Task = System.Threading.Tasks.Task;
-using Microsoft.Web.LibraryManager.Vsix.Contracts;
 
 namespace Microsoft.Web.LibraryManager.Vsix
 {
     internal sealed class RestoreOnBuildCommand
     {
         private bool _isPackageInstalled;
         private readonly IComponentModel _componentModel;
-        private readonly Package _package;
+        private readonly AsyncPackage _package;
         private readonly IDependenciesFactory _dependenciesFactory;
 
-        private RestoreOnBuildCommand(Package package, OleMenuCommandService commandService, IDependenciesFactory dependenciesFactory)
+        private RestoreOnBuildCommand(AsyncPackage package, OleMenuCommandService commandService, IDependenciesFactory dependenciesFactory)
         {
             _package = package;
             _componentModel = VsHelpers.GetService<SComponentModel, IComponentModel>();
             _dependenciesFactory = dependenciesFactory;
 
             var cmdId = new CommandID(PackageGuids.guidLibraryManagerPackageCmdSet, PackageIds.RestoreOnBuild);
-            var cmd = new OleMenuCommand(ExecuteHandlerAsync, cmdId);
-            cmd.BeforeQueryStatus += BeforeQueryStatusHandlerAsync;
+            var cmd = new OleMenuCommand((s, e) => package.JoinableTaskFactory.RunAsync(() => ExecuteAsync(s, e)),
+                                         cmdId);
+            cmd.BeforeQueryStatus += (s, e) => _package.JoinableTaskFactory.RunAsync(() => BeforeQueryStatusAsync(s, e));
```

Comment:
Is this an intentional switch to `_package` (to avoid the capture?) since its being stored in the object? And if so, why not on the previous lambda on line 33?

Reply:
oversight - the previous line should also be `_package`.  Nice catch